x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -234,14 +234,20 @@ export const TakeActionDropdown = React.memo(
     });
 
     const items: React.ReactElement[] = useMemo(
-      () => [
-        ...(tGridEnabled ? addToCaseActionItems : []),
-        ...alertsActionItems,
-        ...hostIsolationActionItems,
-        ...endpointResponseActionsConsoleItems,
-        ...(osqueryAvailable ? [osqueryActionItem] : []),
-        ...investigateInTimelineActionItems,
-      ],
+      () =>
+        [
+          ...(tGridEnabled ? addToCaseActionItems : []),
+          ...alertsActionItems,
+          ...hostIsolationActionItems,
+          ...endpointResponseActionsConsoleItems,
+          ...(osqueryAvailable ? [osqueryActionItem] : []),
+          ...investigateInTimelineActionItems,
+        ].map((element, i) => ({
+          ...element,
+          key: `${
```

Comment:
we were getting the React unique "key" error, so added a fix while I was here

Reply:
Instead of doing this here, can we just add a key to https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/use_add_to_case_actions.tsx#L123 and https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/use_add_to_case_actions.tsx#L132 ? Might also need to add one to https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/components/endpoint_responder/use_responder_action_item.tsx#L47 since that is the top level component in the array, even though the underlying EuiContextMenuItem has one, since that is the outer most component, but don't think we should add keys like this

Reply:
Got it, thanks! https://github.com/elastic/kibana/pull/144249/commits/d0e8a42658d306860680c5e34241e25636506feb